src/module-loader/dotnet/src/MorganStanley.ComposeUI.ModuleLoader.Abstractions/IModuleLoader.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Morgan Stanley makes this available to you under the Apache License,
+// Version 2.0 (the "License"). You may obtain a copy of the License at
+// 
+//      http://www.apache.org/licenses/LICENSE-2.0.
+// 
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership. Unless required by applicable law or agreed
+// to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions
+// and limitations under the License.
+
+namespace MorganStanley.ComposeUI.ModuleLoader;
+
+public interface IModuleLoader
+{
+    void RequestStartProcess(StartupRequest startupRequest);
+
+    void RequestStopProcess(StopRequest stopRequest);
+
+    IObservable<LifetimeEvent> LifetimeEvents { get; }
```

Comment:
How are we going to enumerate module instances? It would make sense to have a `GetInstances` method that returns a collection of `ModuleInstance` objects (to be defined). `ModuleInstance` should hold information like `InstanceId`, `ModuleId`, `ModuleName`, `State` (the lifetime state, like `Starting`, `Started`, etc. mirroring `LifetimeEventType`) and `StartupRequest` (or even the whole `StartupContext`). This is not neeed right now, but it's worth adding a TODO.